mono/metadata/icall-def.h

```diff
@@ -293,6 +293,9 @@ HANDLES(ICALL(ENV_18, "internalGetGacPath", ves_icall_System_Environment_GetGacP
 HANDLES(ICALL(ENV_19, "internalGetHome", ves_icall_System_Environment_InternalGetHome))
 ICALL(ENV_20, "set_ExitCode", mono_environment_exitcode_set)
 
+ICALL_TYPE(EXCEPTION, "System.Exception", EXCEPTION_1)
+HANDLES(ICALL(EXCEPTION_1, "ReportUnhandledException", ves_icall_System_Exception_ReportUnhandledException))
```

Comment:
We'll need corresponding icalls on the IL2CPP side.

Reply:
I already implemented them on a branch in IL2CPP: https://github.cds.internal.unity3d.com/unity/il2cpp/tree/os-synchronization-context

I just need to update Mono on that branch because the new tests I added fail without these icalls.